zeno/include/zeno/utils/binary_reader.h

```diff
@@ -0,0 +1,24 @@
+//
+// Created by zhouhang on 2023/4/28.
+//
+
+#ifndef ZENO_BINARY_READER_H
+#define ZENO_BINARY_READER_H
+#include <vector>
+#include <stdint.h>
+#include "zeno/utils/vec.h"
+
+namespace zeno {
+class BinaryReader {
+    size_t cur = 0;
+    std::vector<uint8_t> data;
+public:
+    BinaryReader(std::vector<uint8_t> data_);
+    uint32_t read_u32_LE();
+    float read_f32_LE();
+    vec4f read_vec4f_LE();
+};
```

Comment:
I would suggest only one method is enough:
```cpp
template <class T>
T read() {
 T &ret = *(T *)(data.data() + cur);
 cur += sizeof(T);
 return ret;
}
```

Reply:
Also, you'd better add bound checks:
```cpp
if (cur + sizeof(T) >= data.size()) throw std::out_of_range("BinaryReader::read");
```

Reply:
must use > rather than >=

Reply:
> 

must use '>' rather than '>='